test/test_irods_http_api.py

```diff
@@ -2339,6 +2339,71 @@ def test_server_correctly_handles_embedded_CRLFs_in_multipart_form_data(self):
             })
             self.logger.debug(r.content)
 
+    def test_server_does_not_segfault_when_parallel_write_init_is_invoked_multiple_times_on_same_logical_path(self):
+        headers = {'Authorization': 'Bearer ' + self.rodsuser_bearer_token}
+        data_object = f'/{self.zone_name}/home/{self.rodsuser_username}/parallel_write_no_segfaults.txt'
+
+        try:
+            # Start a parallel-write operation.
+            r = requests.post(self.url_endpoint, headers=headers, data={
+                'op': 'parallel_write_init',
+                'lpath': data_object,
+                'stream-count': 2
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            parallel_write_handle = result['parallel_write_handle']
+
+            # Make sure the parallel_write_shutdown operation is invoked now that we
+            # have a parallel-write handle.
+            invoke_parallel_write_shutdown = True
```

Comment:
I believe this will need to be defined (and set to `False`, I suppose) before the assertions above. The `finally` block will have errors related to `invoke_parallel_write_shutdown` being used before it's defined. I realize the cleanup would not be necessary in that case, but it's just extra noise we don't need and can easily be avoided, IMO.

Reply:
Agreed.

Copy-paste got me. Will update.

Reply:
Done.